student/views.py

```diff
@@ -1,3 +1,28 @@
-# from django.shortcuts import render
+from django.shortcuts import render, HttpResponseRedirect
+from .models import Student
+from .forms import UpdateStudentAccountSettingsForm
 
-# Create your views here.
+
+def account_update_success(request):
+    context = {}
+    add_is_student_to_context(request, context)
+    return render(request, 'account_settings_update_success.html', context)
+
+
+def update_student_account_settings_view(request):
+    context = {}
+    try:
+        student_object = Student.objects.get(user_id=request.user.id)
+    except Student.DoesNotExist:
+        return render(request, "dead_end.html")
+    update_student_form = UpdateStudentAccountSettingsForm(request.POST or None, instance=student_object)
+    if update_student_form.is_valid():
+        update_student_form.save()
+        return HttpResponseRedirect("/account_update_success/")
```

Comment:
Form handling code typically needs to deal with 3 scenarios:

1. When the form is initially fetched and needs to be populated with initial data from the DB
2. When form data is posted from the browser and its invalid
3. When form data is posted from the browser and its valid

Its not clear how this code deals with those scenarios, especially with the distinction between 1 and 2/3. It seems there is some assumption that the form would be invalid if its initialized from DB data only. This might be causing an unnecessary DB save.

Reply:
I am not sure I understand what you mean here,
about 1, if I managed to get the student object from the DB, why would the form not manage to load?
About 2, the form checks the data automatically and won't let you submit it if the inserted data is invalid.
I also don't understand why the form would be invalid if it's initialized from DB data only..

Reply:
> if I managed to get the student object from the DB, why would the form not manage to load?

What is testing that your code is actually doing that?

>  the form checks the data automatically and won't let you submit it if the inserted data is invalid.

What checks that the view is actually checking for validity and responding properly tro the user instead of, say, throwing a nasty exception on an invalid save attempt?

What you wrote in your last comment are assumptions based on your understanding of how your code and Django works. Tests should not be making such assumptions. They should test that you get the exact behavior you need given controlled inputs without making too many assumptions on the internal implementation. Is there any guarantee that the implementation even uses proper Django Form and View objects? It might change over time. The only thing that can guarantee it works as needed is a good test.

Reply:
I think you mean that I add these tests in the tests file, but this is the views file so I don't understand if anything else is needed here or not.

Reply:
Ok with regard to the code clarity - for example its hard to see the difference between the behavior on POST and on GET because there is explicit IF for that

Reply:
Fixed.